File: apps/inference-service/backend/app/schemas/vehicle.py
from pydantic import BaseModel, Field
from typing import Optional, Dict

class VehicleCreateIn(BaseModel):
    vehicleType: str
    vehicleModel: str
    registrationDate: str  # YYYY-MM-DD
    plateNo: str = Field(..., min_length=3)

class VehicleOut(BaseModel):
    id: str
    ownerId: str
    vehicleType: str
    vehicleModel: str
    registrationDate: str
    plateNo: str
    images: Dict[str, Optional[str]]

File: apps/mobile/lib/models/vehicle.dart
class Vehicle {
  final String id;
  final String ownerId;
  final String vehicleType;
  final String vehicleModel;
  final String registrationDate;
  final String plateNo;
  final Map<String, dynamic> images;

  Vehicle({
    required this.id,
    required this.ownerId,
    required this.vehicleType,
    required this.vehicleModel,
    required this.registrationDate,
    required this.plateNo,
    required this.images,
  });

  factory Vehicle.fromJson(Map<String, dynamic> j) => Vehicle(
    id: j['id'],
    ownerId: j['ownerId'],
    vehicleType: j['vehicleType'],
    vehicleModel: j['vehicleModel'],
    registrationDate: j['registrationDate'],
    plateNo: j['plateNo'],
    images: Map<String, dynamic>.from(j['images'] ?? {}),
  );
}
